os/arch/arm/src/common/up_createstack.c

```diff
@@ -198,14 +198,27 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		 * in user threads and null for kernel threads.
 		 */
 		if (!uheap) {
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
+#ifdef CONFIG_ARMV8M_MPU
+			tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(MPU_ALIGNMENT_BYTES, stack_alloc_size);
+#else
+			tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(CONFIG_MPU_STACK_GUARD_SIZE, stack_alloc_size);
+#endif
+#else
 			tcb->stack_alloc_ptr = (uint32_t *)kmm_malloc(stack_alloc_size);
+#endif
 		} else
 #endif
 		{
 			/* Use the user-space allocator if this is a task or pthread */
-#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
 			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
+#ifdef CONFIG_ARMV8M_MPU
```

Comment:
I want to remove duplicated codes.
How about below?
```c
#ifdef CONFIG_ARMV8M_MPU
#define STACK_PROTCTION_MPU_ALIGNMENT (32) // fixed size by ARMv8-M spec
#else // CONFIG_ARMV7M_MPU
#define STACK_PROTCTION_MPU_ALIGNMENT (CONFIG_MPU_STACK_GUARD_SIZE) // In ARMv7-M, MPU requires alignment with? for? to? region size
#endif

#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
    stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GURAD_SIZE;
    tcb->stack_alloc_ptr = (uint32_t *)kumm_memalign(STACK_PROTCTION_MPU_ALIGNMENT, stack_alloc_size);
#else 
    tcb->stack_alloc_ptr = (uint32_t *)kumm_malloc(stack_size);
#endif 
```
Please leave the comment to say why it has different value for alignment.

Reply:
Done. Please check.